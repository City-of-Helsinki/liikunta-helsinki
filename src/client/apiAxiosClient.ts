import axios from "axios";

import Config from "../config";

const apiAxiosClient = axios.create({
  baseURL: Config.nextApiEndpoint,
});

export default apiAxiosClient;
