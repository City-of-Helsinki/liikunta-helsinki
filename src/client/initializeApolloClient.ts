// https://www.apollographql.com/blog/building-a-next-js-app-with-apollo-client-slash-graphql/
export default function initializeApolloClient(
  initialState = null,
  cachedClient,
  createClient
) {
  const _apolloClient = cachedClient ?? createClient();

  // Initial state hydration
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = _apolloClient.extract();

    // Restore the cache using the data passed from
    // getStaticProps/getServerSideProps combined with the existing cached data
    _apolloClient.cache.restore({ ...existingCache, ...initialState });
  }

  // For SSG and SSR always create a new Apollo Client
  if (typeof window === "undefined") {
    return _apolloClient;
  }

  // Create the Apollo Client once in the client
  if (!cachedClient) {
    cachedClient = _apolloClient;
  }

  return _apolloClient;
}
