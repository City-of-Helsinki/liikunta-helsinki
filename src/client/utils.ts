import { Connection, MenuItem } from "../types";

export function getNodes<T>(connection: Connection<T>): T[] {
  return connection.edges.map(({ node }) => node);
}

export function sortMenuItems(menuItemsConnection: { nodes: MenuItem[] }) {
  const menuItems = menuItemsConnection.nodes;
  const sortedMenuItems = [...menuItems].sort((a, b) => a.order - b.order);

  return {
    ...menuItemsConnection,
    nodes: sortedMenuItems,
  };
}

export function getQlLanguage(language: string) {
  return {
    fi: "FI",
    sv: "SV",
    en: "EN",
  }[language];
}

export function getUnifiedSearchLanguage(language: string) {
  return {
    fi: "FINNISH",
    sv: "SWEDISH",
    en: "ENGLISH",
  }[language];
}

export function getMenuLocationFromLanguage(language: string) {
  switch (language) {
    case "fi":
      return "PRIMARY";
    case "sv":
      return "PRIMARY___SV";
    case "en":
      return "PRIMARY___EN";
    default:
      return "PRIMARY";
  }
}
