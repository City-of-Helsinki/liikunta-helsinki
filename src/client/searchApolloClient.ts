import {
  ApolloClient,
  NormalizedCacheObject,
  InMemoryCache,
} from "@apollo/client";
import { relayStylePagination } from "@apollo/client/utilities";

import Config from "../config";

const cache: InMemoryCache = new InMemoryCache({
  typePolicies: {
    Query: {
      fields: {
        unifiedSearch: relayStylePagination(),
      },
    },
  },
});

const searchApolloClient: ApolloClient<NormalizedCacheObject> = new ApolloClient(
  {
    cache,
    uri: Config.unifiedSearchGraphqlEndpoint,
    ssrMode: false,
  }
);

export default searchApolloClient;
