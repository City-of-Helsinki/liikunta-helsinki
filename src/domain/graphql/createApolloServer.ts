import {
  ApolloServer,
  Config as ApolloServerConfig,
} from "apollo-server-micro";
import {
  ApolloServerPluginLandingPageGraphQLPlayground,
  ApolloServerPluginLandingPageDisabled,
  Context,
  ContextFunction,
} from "apollo-server-core";
import accepts from "accepts";
import { NextApiRequest } from "next";
import { DocumentNode, gql } from "@apollo/client";

import Config from "../../config";
import createVenueSchema from "./venue/venueSchema";
import venueQueryResolver from "./venue/venueQueryResolver";
import venuesByIdsResolver from "./venue/venuesByIdsResolver";
import Venue from "./venue/venueResolver";
import eventSchema from "./event/eventSchema";
import Event from "./event/eventResolver";
import Hauki from "./services/HaukiDataSource";
import Tprek from "./services/TprekDataSource";
import Linked from "./services/linked/LinkedDataSource";
import LinkedPaginatedConnectionResolver from "./services/linked/LinkedPaginatedConnectionResolver";
import LiikuntaLoggerPlugin from "./LiikuntaLoggerPlugin";
import eventsQueryResolver from "./event/eventsQueryResolver";
import paginationSchema from "./paginationSchema";

// Note: In the current version of GraphQL, you can’t have an empty type even if
// you intend to extend it later. So we need to make sure the Query type has at
// least one field — in this case we can add a fake _empty field. Hopefully in
// future versions it will be possible to have an empty type to be extended
// later.
// https://www.apollographql.com/blog/backend/schema-design/modularizing-your-graphql-schema-code/
const initQueryTypeDefs = gql`
  type Query {
    _empty: String
  }
`;

const typeDefs = (liikuntaServerConfig?: LiikuntaServerConfig) => [
  initQueryTypeDefs,
  paginationSchema,
  createVenueSchema(liikuntaServerConfig),
  eventSchema,
];
const dataSourcesFactory = () => ({
  hauki: new Hauki(),
  tprek: new Tprek(),
  linked: new Linked(),
});

const resolvers = {
  Query: {
    venue: venueQueryResolver,
    venuesByIds: venuesByIdsResolver,
    events: eventsQueryResolver,
  },
  Venue,
  EventsConnection: LinkedPaginatedConnectionResolver,
  Event,
};

function acceptsLanguages(
  req: NextApiRequest,
  languages: string[]
): typeof languages[number] | false {
  const accept = accepts(req);

  return accept.languages(languages);
}

function handleContext({ req }) {
  const language = acceptsLanguages(req, Config.locales);

  return {
    language,
  };
}

const plugins = [
  new LiikuntaLoggerPlugin(),
  process.env.NODE_ENV === "production"
    ? ApolloServerPluginLandingPageDisabled
    : ApolloServerPluginLandingPageGraphQLPlayground(),
];

export type LiikuntaServerConfig = {
  haukiEnabled?: boolean;
};

type ServerConfig = Omit<ApolloServerConfig, "typDefs"> &
  LiikuntaServerConfig & {
    typeDefs?: (
      liikuntaServerConfig?: LiikuntaServerConfig
    ) => DocumentNode | Array<DocumentNode> | string | Array<string>;
  };

export default function createApolloServer(apolloServerConfig?: ServerConfig) {
  const injectIntoContext = (
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    context: Context | ContextFunction<any>,
    injectedObject: LiikuntaServerConfig
  ) => {
    if (typeof context === "function") {
      return (params) => ({
        ...context(params),
        ...injectedObject,
      });
    }

    return {
      ...context,
      ...injectedObject,
    };
  };
  const liikuntaServerConfig = {
    haukiEnabled: apolloServerConfig?.haukiEnabled,
  };

  return new ApolloServer({
    dataSources: apolloServerConfig?.dataSources ?? dataSourcesFactory,
    typeDefs: (apolloServerConfig?.typeDefs ?? typeDefs)(liikuntaServerConfig),
    resolvers: apolloServerConfig?.resolvers ?? resolvers,
    // Uncomment line below to enable apollo tracing
    // plugins: [require('apollo-tracing').plugin()]
    context: injectIntoContext(
      apolloServerConfig?.context ?? handleContext,
      liikuntaServerConfig
    ),
    plugins: apolloServerConfig?.plugins ?? plugins,
    // Turn off introspection in production because this endpoint is meant
    // for the liikunta site application only
    introspection: process.env.NODE_ENV !== "production",
  });
}
