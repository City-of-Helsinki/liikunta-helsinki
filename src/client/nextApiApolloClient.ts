import https from "https";

import {
  ApolloClient,
  InMemoryCache,
  NormalizedCacheObject,
  HttpLink,
  HttpOptions,
} from "@apollo/client";
import { useMemo } from "react";

import Config from "../config";
import { logger } from "../domain/logger";
import initializeApolloClient from "./initializeApolloClient";

let apiApolloClient: ApolloClient<NormalizedCacheObject>;

function getHttpLink(uri: string) {
  let options: HttpOptions = {
    uri,
  };

  // Our review environment can't provide a valid certificate. Hence we allow
  // the application to be configured so that unauthorized requests are not
  // rejected. This allows us to test API changes in the review environment.
  if (Config.allowUnauthorizedRequests && new URL(uri).protocol === "https:") {
    logger.info("Allowing unauthorized requests");
    options = {
      ...options,
      fetchOptions: {
        agent: new https.Agent({ rejectUnauthorized: false }),
      },
    };
  }

  return new HttpLink(options);
}

function createNextApiApolloClient() {
  return new ApolloClient({
    cache: new InMemoryCache({}),
    link: getHttpLink(Config.nextApiGraphqlEndpoint),
    ssrMode: !process.browser,
  });
}

export default function initializeNextApiApolloClient(initialState = null) {
  return initializeApolloClient(
    initialState,
    apiApolloClient,
    createNextApiApolloClient
  );
}

export function useNextApiApolloClient(initialState) {
  const store = useMemo(() => initializeNextApiApolloClient(initialState), [
    initialState,
  ]);

  return store;
}
