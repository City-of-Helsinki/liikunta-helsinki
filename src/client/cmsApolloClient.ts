import { InMemoryCache } from "@apollo/client";
import { useMemo } from "react";

import Config from "../config";
import initializeApolloClient from "./initializeApolloClient";
import LiikuntaApolloClient from "./LiikuntaApolloClient";
import { sortMenuItems } from "./utils";

let cmsApolloClient: LiikuntaApolloClient;

function createCmsApolloClient() {
  return new LiikuntaApolloClient({
    ssrMode: !process.browser,
    uri: Config.cmsGraphqlEndpoint,
    cache: new InMemoryCache({
      typePolicies: {
        RootQuery: {
          queryType: true,
        },
        MenuItems: {
          fields: {
            nodes: {
              read(nodes) {
                return sortMenuItems(nodes);
              },
            },
          },
        },
      },
    }),
  });
}

export default function initializeCmsApollo(initialState = null) {
  return initializeApolloClient(
    initialState,
    cmsApolloClient,
    createCmsApolloClient
  );
}

export function useCmsApollo(initialState) {
  const store = useMemo(() => initializeCmsApollo(initialState), [
    initialState,
  ]);

  return store;
}
