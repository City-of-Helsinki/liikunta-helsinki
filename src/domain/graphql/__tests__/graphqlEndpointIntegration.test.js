import { gql } from "@apollo/client";

import createApolloServer from "../createApolloServer";

const server = createApolloServer({
  context: () => {
    return {
      language: "fi",
    };
  },
  plugins: [],
});

beforeAll(() => {
  jest.useFakeTimers();
  jest.setSystemTime(new Date(2021, 11, 13));
});

afterAll(() => {
  jest.useRealTimers();
});

const GET_VENUE = gql`
  query GetVenueTestQuery($id: ID!) {
    venue(id: $id) {
      addressLocality
      dataSource
      description
      email
      id
      isOpen
      image
      infoUrl
      name
      accessibilitySentences {
        groupName
        sentences
      }
      openingHours {
        date
        times {
          startTime
          endTime
          endTimeOnNextDay
          resourceState
          fullDay
        }
      }
      position {
        type
        coordinates
      }
      postalCode
      streetAddress
      telephone
      ontologyWords {
        id
        label
      }
      ontologyTree {
        id
        label
      }
    }
  }
`;

test("venue(id) should return a complete venue", async () => {
  const res = await server.executeOperation({
    query: GET_VENUE,
    variables: { id: "tprek:1234" },
  });

  expect(res).toMatchSnapshot();
});

test("venue(id) with incorrect id should error", async () => {
  const res = await server.executeOperation({
    query: GET_VENUE,
    variables: { id: "notsupported:1234" },
  });

  expect(res).toMatchSnapshot();
});

const GET_VENUES_BY_IDS = gql`
  query GetVenueTestQuery($ids: [ID!]!) {
    venuesByIds(ids: $ids) {
      addressLocality
      dataSource
      description
      email
      id
      isOpen
      image
      infoUrl
      name
      accessibilitySentences {
        groupName
        sentences
      }
      openingHours {
        date
        times {
          startTime
          endTime
          endTimeOnNextDay
          resourceState
          fullDay
        }
      }
      position {
        type
        coordinates
      }
      postalCode
      streetAddress
      telephone
      ontologyWords {
        id
        label
      }
      ontologyTree {
        id
        label
      }
    }
  }
`;

test("venuesByIds(ids) should return matching venues", async () => {
  const res = await server.executeOperation({
    query: GET_VENUES_BY_IDS,
    variables: { ids: ["tprek:1234", "tprek:1235"] },
  });

  expect(res).toMatchSnapshot();
});
