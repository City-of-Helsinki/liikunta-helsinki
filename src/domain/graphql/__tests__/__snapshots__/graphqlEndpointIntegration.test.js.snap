// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`venue(id) should return a complete venue 1`] = `
Object {
  "data": Object {
    "venue": Object {
      "accessibilitySentences": Array [
        Object {
          "groupName": "Reitti pääsisäänkäynnille",
          "sentences": Array [
            "Kulkureitti sisäänkäynnille on opastettu, epätasainen ja kapea sekä valaistu.",
            "Kulkureitillä on 1 porrasaskelma.",
          ],
        },
        Object {
          "groupName": "Pääsisäänkäynti",
          "sentences": Array [
            "Sisäänkäynti on vaikeasti hahmotettava mutta valaistu.",
            "Sisäänkäynnin yhteydessä on 1 porrasaskelma.",
            "Sisäänkäynnin ovet erottuvat selkeästi. Oven ulkopuolella on riittävästi vapaata tilaa liikkumiselle esim. pyörätuolin kanssa. Ovi vaatii summerin painamista, jonka jälkeen ovi aukeaa mutta on raskas.",
            "Sisäänkäynnin yhteydessä on yli 2 cm korkeita kynnyksiä.",
            "Tuulikaappi on ahdas.",
          ],
        },
        Object {
          "groupName": "Sisätilat",
          "sentences": Array [
            "Toimipisteessä on 2 kerrosta.",
            "Tilan keskeisillä kulkureiteillä on vähintään 4 kääntyvää porrasaskelmaa, jossa on käsijohteet molemmilla puolilla.",
            "Sisätilojen ovet erottuvat selkeästi.",
            "Toimipisteessä on esteettömäksi merkitty wc kerroksessa 1. Wc:ssä ei ole tarpeeksi tilaa pyörätuolille.",
            "Toinen esteettömäksi merkitty wc on kerroksessa 0.",
          ],
        },
      ],
      "addressLocality": "Helsinki",
      "connections": Array [
        Object {
          "name": "Päiväkodinjohtaja",
          "phone": "+358 9 310 64132",
          "sectionType": "PHONE_OR_EMAIL",
        },
        Object {
          "name": "Hakemus kerhotoimintaan, esiopetukseen ja esiopetusta täydentävään varhaiskasvatukseen",
          "phone": null,
          "sectionType": "ESERVICE_LINK",
        },
        Object {
          "name": "Kerhohakemus sähköisessä asioinnissa",
          "phone": null,
          "sectionType": "ESERVICE_LINK",
        },
        Object {
          "name": "Varhaiskasvatushakemus Asti-asiointipalvelussa",
          "phone": null,
          "sectionType": "ESERVICE_LINK",
        },
      ],
      "dataSource": "tprek",
      "description": null,
      "email": "pk.lappi@hel.fi",
      "id": "tprek:1234",
      "image": null,
      "infoUrl": "https://www.hel.fi/varhaiskasvatus/fi/paivakodit/lappi",
      "isOpen": true,
      "name": "Päiväkoti Lappi",
      "ontologyTree": Array [
        Object {
          "id": 870,
          "label": "Suomenkielinen päivähoito",
        },
        Object {
          "id": 871,
          "label": "päivähoito",
        },
        Object {
          "id": 874,
          "label": "ympärivuorokautinen hoito",
        },
        Object {
          "id": 879,
          "label": "esiopetus",
        },
        Object {
          "id": 1090,
          "label": "Päivähoidon järjestämä esiopetus",
        },
        Object {
          "id": 2125,
          "label": "Päivähoidon järjestämä esiopetus",
        },
      ],
      "ontologyWords": Array [
        Object {
          "id": 160,
          "label": "esiopetus (päivähoidon järjestämä)",
        },
        Object {
          "id": 663,
          "label": "suomenkielinen päivähoito",
        },
        Object {
          "id": 831,
          "label": "ympärivuorokautinen hoito (lasten päivähoito)",
        },
        Object {
          "id": 861,
          "label": "suomenkielinen esiopetus",
        },
      ],
      "openingHours": Array [
        Object {
          "date": "2021-12-13",
          "times": Array [],
        },
        Object {
          "date": "2021-12-14",
          "times": Array [],
        },
        Object {
          "date": "2021-12-15",
          "times": Array [],
        },
        Object {
          "date": "2021-12-16",
          "times": Array [],
        },
        Object {
          "date": "2021-12-17",
          "times": Array [],
        },
        Object {
          "date": "2021-12-18",
          "times": Array [],
        },
        Object {
          "date": "2021-12-19",
          "times": Array [],
        },
      ],
      "position": Object {
        "coordinates": Array [
          24.926613,
          60.16692,
        ],
        "type": "Point",
      },
      "postalCode": "00180",
      "streetAddress": "Lapinlahdenkatu 6 G",
      "telephone": "+358 9 310 64132",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`venue(id) with incorrect id should error 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Invalid ID parameter],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`venuesByIds(ids) should return matching venues 1`] = `
Object {
  "data": Object {
    "venuesByIds": Array [
      Object {
        "accessibilitySentences": Array [
          Object {
            "groupName": "Reitti pääsisäänkäynnille",
            "sentences": Array [
              "Kulkureitti sisäänkäynnille on opastettu, epätasainen ja kapea sekä valaistu.",
              "Kulkureitillä on 1 porrasaskelma.",
            ],
          },
          Object {
            "groupName": "Pääsisäänkäynti",
            "sentences": Array [
              "Sisäänkäynti on vaikeasti hahmotettava mutta valaistu.",
              "Sisäänkäynnin yhteydessä on 1 porrasaskelma.",
              "Sisäänkäynnin ovet erottuvat selkeästi. Oven ulkopuolella on riittävästi vapaata tilaa liikkumiselle esim. pyörätuolin kanssa. Ovi vaatii summerin painamista, jonka jälkeen ovi aukeaa mutta on raskas.",
              "Sisäänkäynnin yhteydessä on yli 2 cm korkeita kynnyksiä.",
              "Tuulikaappi on ahdas.",
            ],
          },
          Object {
            "groupName": "Sisätilat",
            "sentences": Array [
              "Toimipisteessä on 2 kerrosta.",
              "Tilan keskeisillä kulkureiteillä on vähintään 4 kääntyvää porrasaskelmaa, jossa on käsijohteet molemmilla puolilla.",
              "Sisätilojen ovet erottuvat selkeästi.",
              "Toimipisteessä on esteettömäksi merkitty wc kerroksessa 1. Wc:ssä ei ole tarpeeksi tilaa pyörätuolille.",
              "Toinen esteettömäksi merkitty wc on kerroksessa 0.",
            ],
          },
        ],
        "addressLocality": "Helsinki",
        "connections": Array [
          Object {
            "name": "Päiväkodinjohtaja",
            "phone": "+358 9 310 64132",
            "sectionType": "PHONE_OR_EMAIL",
          },
          Object {
            "name": "Hakemus kerhotoimintaan, esiopetukseen ja esiopetusta täydentävään varhaiskasvatukseen",
            "phone": null,
            "sectionType": "ESERVICE_LINK",
          },
          Object {
            "name": "Kerhohakemus sähköisessä asioinnissa",
            "phone": null,
            "sectionType": "ESERVICE_LINK",
          },
          Object {
            "name": "Varhaiskasvatushakemus Asti-asiointipalvelussa",
            "phone": null,
            "sectionType": "ESERVICE_LINK",
          },
        ],
        "dataSource": "tprek",
        "description": null,
        "email": "pk.lappi@hel.fi",
        "id": "tprek:1234",
        "image": null,
        "infoUrl": "https://www.hel.fi/varhaiskasvatus/fi/paivakodit/lappi",
        "isOpen": true,
        "name": "Päiväkoti Lappi",
        "ontologyTree": Array [
          Object {
            "id": 870,
            "label": "Suomenkielinen päivähoito",
          },
          Object {
            "id": 871,
            "label": "päivähoito",
          },
          Object {
            "id": 874,
            "label": "ympärivuorokautinen hoito",
          },
          Object {
            "id": 879,
            "label": "esiopetus",
          },
          Object {
            "id": 1090,
            "label": "Päivähoidon järjestämä esiopetus",
          },
          Object {
            "id": 2125,
            "label": "Päivähoidon järjestämä esiopetus",
          },
        ],
        "ontologyWords": Array [
          Object {
            "id": 160,
            "label": "esiopetus (päivähoidon järjestämä)",
          },
          Object {
            "id": 663,
            "label": "suomenkielinen päivähoito",
          },
          Object {
            "id": 831,
            "label": "ympärivuorokautinen hoito (lasten päivähoito)",
          },
          Object {
            "id": 861,
            "label": "suomenkielinen esiopetus",
          },
        ],
        "openingHours": Array [
          Object {
            "date": "2021-12-13",
            "times": Array [],
          },
          Object {
            "date": "2021-12-14",
            "times": Array [],
          },
          Object {
            "date": "2021-12-15",
            "times": Array [],
          },
          Object {
            "date": "2021-12-16",
            "times": Array [],
          },
          Object {
            "date": "2021-12-17",
            "times": Array [],
          },
          Object {
            "date": "2021-12-18",
            "times": Array [],
          },
          Object {
            "date": "2021-12-19",
            "times": Array [],
          },
        ],
        "position": Object {
          "coordinates": Array [
            24.926613,
            60.16692,
          ],
          "type": "Point",
        },
        "postalCode": "00180",
        "streetAddress": "Lapinlahdenkatu 6 G",
        "telephone": "+358 9 310 64132",
      },
      Object {
        "accessibilitySentences": Array [
          Object {
            "groupName": "Reitti pääsisäänkäynnille",
            "sentences": Array [
              "Kulkureitti sisäänkäynnille on opastettu, epätasainen ja kapea sekä valaistu.",
              "Kulkureitillä on 1 porrasaskelma.",
            ],
          },
          Object {
            "groupName": "Pääsisäänkäynti",
            "sentences": Array [
              "Sisäänkäynti on vaikeasti hahmotettava mutta valaistu.",
              "Sisäänkäynnin yhteydessä on 1 porrasaskelma.",
              "Sisäänkäynnin ovet erottuvat selkeästi. Oven ulkopuolella on riittävästi vapaata tilaa liikkumiselle esim. pyörätuolin kanssa. Ovi vaatii summerin painamista, jonka jälkeen ovi aukeaa mutta on raskas.",
              "Sisäänkäynnin yhteydessä on yli 2 cm korkeita kynnyksiä.",
              "Tuulikaappi on ahdas.",
            ],
          },
          Object {
            "groupName": "Sisätilat",
            "sentences": Array [
              "Toimipisteessä on 2 kerrosta.",
              "Tilan keskeisillä kulkureiteillä on vähintään 4 kääntyvää porrasaskelmaa, jossa on käsijohteet molemmilla puolilla.",
              "Sisätilojen ovet erottuvat selkeästi.",
              "Toimipisteessä on esteettömäksi merkitty wc kerroksessa 1. Wc:ssä ei ole tarpeeksi tilaa pyörätuolille.",
              "Toinen esteettömäksi merkitty wc on kerroksessa 0.",
            ],
          },
        ],
        "addressLocality": "Helsinki",
        "connections": Array [
          Object {
            "name": "Päiväkodinjohtaja",
            "phone": "+358 9 310 64132",
            "sectionType": "PHONE_OR_EMAIL",
          },
          Object {
            "name": "Hakemus kerhotoimintaan, esiopetukseen ja esiopetusta täydentävään varhaiskasvatukseen",
            "phone": null,
            "sectionType": "ESERVICE_LINK",
          },
          Object {
            "name": "Kerhohakemus sähköisessä asioinnissa",
            "phone": null,
            "sectionType": "ESERVICE_LINK",
          },
          Object {
            "name": "Varhaiskasvatushakemus Asti-asiointipalvelussa",
            "phone": null,
            "sectionType": "ESERVICE_LINK",
          },
        ],
        "dataSource": "tprek",
        "description": null,
        "email": "pk.lappi@hel.fi",
        "id": "tprek:1235",
        "image": null,
        "infoUrl": "https://www.hel.fi/varhaiskasvatus/fi/paivakodit/lappi",
        "isOpen": true,
        "name": "Päiväkoti Lappi",
        "ontologyTree": Array [
          Object {
            "id": 870,
            "label": "Suomenkielinen päivähoito",
          },
          Object {
            "id": 871,
            "label": "päivähoito",
          },
          Object {
            "id": 874,
            "label": "ympärivuorokautinen hoito",
          },
          Object {
            "id": 879,
            "label": "esiopetus",
          },
          Object {
            "id": 1090,
            "label": "Päivähoidon järjestämä esiopetus",
          },
          Object {
            "id": 2125,
            "label": "Päivähoidon järjestämä esiopetus",
          },
        ],
        "ontologyWords": Array [
          Object {
            "id": 160,
            "label": "esiopetus (päivähoidon järjestämä)",
          },
          Object {
            "id": 663,
            "label": "suomenkielinen päivähoito",
          },
          Object {
            "id": 831,
            "label": "ympärivuorokautinen hoito (lasten päivähoito)",
          },
          Object {
            "id": 861,
            "label": "suomenkielinen esiopetus",
          },
        ],
        "openingHours": Array [
          Object {
            "date": "2021-12-13",
            "times": Array [],
          },
          Object {
            "date": "2021-12-14",
            "times": Array [],
          },
          Object {
            "date": "2021-12-15",
            "times": Array [],
          },
          Object {
            "date": "2021-12-16",
            "times": Array [],
          },
          Object {
            "date": "2021-12-17",
            "times": Array [],
          },
          Object {
            "date": "2021-12-18",
            "times": Array [],
          },
          Object {
            "date": "2021-12-19",
            "times": Array [],
          },
        ],
        "position": Object {
          "coordinates": Array [
            24.926613,
            60.16692,
          ],
          "type": "Point",
        },
        "postalCode": "00180",
        "streetAddress": "Lapinlahdenkatu 6 G",
        "telephone": "+358 9 310 64132",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
